Sources/Target/Linux/Thread.cpp

```diff
@@ -186,10 +187,17 @@ void Thread::updateState() {
     break;
 
   case Host::Linux::kProcStateTraced:
-  case Host::Linux::kProcStateStopped:
+  case Host::Linux::kProcStateStopped: {
+    if (_state == kRunning) {
```

Comment:
Having to manually call `::waitpid` here is not super great. Can you explain why we need to do this again? I think I remember some code uses `_stopInfo.reason` and `_stopInfo.event` when being called from `enumerateThreads`. If we want this code to be correct, we need to do this `::waitpid` every time there's a state change from `kRunning` to `kStopped`, or `kTerminated`. Here we only `updateStopInfo` for the first case.